File: 1 to 100/83_remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        visited = []
        current = head
        prev = ListNode(0)
        while current:
            if current.val not in visited:
                visited.append(current.val)
                prev = current
                current = current.next
            else:
                prev.next = current.next
                current = prev.next
        return head
        

File: 1 to 100/C++/83_remove-duplicates-from-sorted-list.cpp
#include<bits/stdc++.h>
using namespace std;
#define endl '\n'
#define int long long
const int mod = 1e9+7;

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* temp = head;
        ListNode* node = NULL;
        ListNode* del = NULL;
        while(temp!=NULL){
            node = temp;
            temp = temp->next;
            if(temp != NULL && temp->val == node->val){
                node->next = temp->next;
                temp->next = NULL;
                del = temp;
                temp = node;
                delete del;
            }
        }
        return head;
    }
};
signed main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);cout.tie(0);
    Solution obj;
}